lib/sim.c

```diff
@@ -907,6 +923,34 @@ sim_generate_parents(sim_t *self)
             goto out;
         }
     }
+    /* In case of a recombination event in the Kingman phase
+     * we require exactly one recombination point.
+     */
+    if (arg_flag == 1) {
+        node = C[0]->ancestry.head;
+        locus_mapping = (int_map_value_t *) node->item;
+        site_2 = (double) locus_mapping->key;
+        node = node->next;
+				total_rate = 0.0;
+        while (node != NULL) {
+            site_1 = site_2;
+            locus_mapping = (int_map_value_t *) node->item;
+            site_2 = (double) locus_mapping->key;
+            recomb_rates[i] = site_2 - site_1;
+            total_rate += recomb_rates[i];
+            i++;
+            node = node->next;
+        } 
```

Comment:
So, what we're doing here is iterating over all the mapped loci in this individual, and calculating the probability that a recombination occurs between two pairs, right? I think it would be good to phrase this explicitly as probablilities, and reuse the ``probability_list_select`` function above to choose the location we're interested in.